Marlin/src/module/stepper.cpp

```diff
@@ -2136,6 +2137,7 @@ void Stepper::report_positions() {
       _APPLY_DIR(AXIS, _INVERT_DIR(AXIS)^DIR^INVERT);   \
       _PULSE_WAIT;                                      \
```

Comment:
I've always wondered whether it's really needed to wait after setting the direction. The direction pin change should occur at close to light speed and certainly should take effect before the pulse starts.

Reply:
```cpp
#define BABYSTEP_AXIS(AXIS, INVERT, DIR) {            \
    const uint8_t old_dir = _READ_DIR(AXIS);          \
    _ENABLE(AXIS);                                    \
    _APPLY_DIR(AXIS, _INVERT_DIR(AXIS)^DIR^INVERT);   \
    _SAVE_START;                                      \
    _APPLY_STEP(AXIS)(!_INVERT_STEP_PIN(AXIS), true); \
    _PULSE_WAIT;                                      \
    _APPLY_STEP(AXIS)(_INVERT_STEP_PIN(AXIS), true);  \
    _APPLY_DIR(AXIS, old_dir);                        \
  }
```

Reply:
There are specific constraints for the delay between dir and step, 200ns for a4988, 650ns for drv8825, this doesn't seem to respect that? this may not be an issue for slow MCUs but will be for some.

Reply:
Nothing in the history of Marlin has ever bothered to insert the ideal delay after setting the direction pin, or to distinguish between A4988 and DRV8825 stepper drivers. Perhaps now would be a good moment to do that since we now have more accurate delay options available.

Reply:
> Nothing in the history of Marlin has ever bothered to insert the ideal delay after setting the direction pin

It's understandable that the code was designed to take advantage of the MCU performance (or lack thereof) I think the Dir change wait is an important thing to adhere to Enable wait not so much because of how Marlin handles that, pulse duration was the important one and we have that covered already.

Reply:
On AVR a delay of 200ns is covered by 3.2 NOPs. The copy of a 16-bit integer from SRAM to a register probably takes that amount of time, maybe a bit more. To get a guaranteed 650ns delay will take some additional NOPs.